### PR TITLE
Studio/Agent/Server Integration

### PR DESCRIPTION
**Pull request by openbots-ff**
- Added commands for Assets, Credentials and QueueItems.
- Server Credentials/Agent Id are accessed via a Registry Key/Environment Variable respectively
- Publishing a project will create a Process in the Server

### CODE REVIEW

File: OpenBots.Core/Server/API Methods/ProcessMethods.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+namespace OpenBots.Core.Server.API_Methods
+{
+    public class ProcessMethods
+    {
+        public static void UploadProcess(RestClient client, string name, string filePath)
+        {
+            var request = new RestRequest("api/v1/Processes", Method.POST);
+            request.AddParameter("Name", name);
+            request.AddParameter("Status", "Published");
+            request.AddParameter("OrganizationId", "ef8a6670-f522-4fcd-a55e-90aa92d1deb7"); //TODO Remove once API is updated to not require it
```
Comment (openbots-ch) [inline OpenBots.Core/Server/API Methods/ProcessMethods.cs:13]: Lets prioritize this removal before merging.

File: OpenBots.Core/Server/Interfaces/INamedEntity.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace OpenBots.Core.Server.Interfaces
+{
+    public interface INamedEntity : IEntity, INameIDPair
+    {
+
```
Comment (openbots-ch) [inline OpenBots.Core/Server/Interfaces/INamedEntity.cs:5]: Lets leave a note to describe the intent for the file (What purpose it serves). In this case its just an extension point for IEntity, correct?

File: OpenBots.Studio/UI/Forms/Supplement Forms/frmPublishProject.cs [hunk 3]
```diff
@@ -98,7 +99,19 @@ private bool PublishProject()
...
+
+                NotificationMessage = $"'{_projectName}' published successfully";
+
+                try {
+                    var client = AuthMethods.GetAuthToken();
+                    ProcessMethods.UploadProcess(client, _projectName, nugetFilePath);
+                }
+                catch (Exception)
+                {
+                    NotificationMessage = $"'{_projectName}' was published locally but was unable to connect to the server";
```
Comment (openbots-ch) [inline OpenBots.Studio/UI/Forms/Supplement Forms/frmPublishProject.cs:111]: Lets reword this to not assume a connection is attended. Ex: "{_projectName}' was published locally. To publish to an OpenBots Server please install and connect the OpenBots Agent."

**Review by openbots-ch — CHANGES_REQUESTED** [review]

File: OpenBots.Commands/OpenBots.Commands.QueueItem/AddQueueItemCommand.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                QueueId = queue.Id,
+                Type = v_QueueItemType,
+                JsonType = vJsonType,
+                DataJson = vQueueItemTextValue,
+                Name = vQueueItemName,
+                IsDeleted = false,
+                Priority = int.Parse(vPriority)
+            };
+
+            QueueItemMethods.EnqueQueueItem(client, queueItem);
```
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.QueueItem/AddQueueItemCommand.cs:104]: Done

File: OpenBots.Core/Server/API Methods/AuthMethods.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+using System.IO;
+using System.Net.Http;
+
+namespace OpenBots.Core.Server.API_Methods
+{
+    public class AuthMethods
+    {
+        public static RestClient GetAuthToken()
+        {
+            var client = new RestClient("https://openbotsserver-dev.azurewebsites.net/");
```
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Core/Server/API Methods/AuthMethods.cs:16]: replaced with URL in settings file

File: OpenBots.Core/Server/API Methods/ProcessMethods.cs (hunk 1 above)
Comment (openbots-ff) [reply OpenBots.Core/Server/API Methods/ProcessMethods.cs:13]: Nicole's changes are sitting in a PR

File: OpenBots.Core/Server/Interfaces/INamedEntity.cs (hunk 2 above)
Comment (openbots-ff) [reply OpenBots.Core/Server/Interfaces/INamedEntity.cs:5]: These are just the original server viewmodels that I'm using to deserialize the API responses. I don't think it serves any purpose aside from being a bridge to the other interfaces.

File: OpenBots.Studio/UI/Forms/Supplement Forms/frmPublishProject.cs (hunk 3 above)
Comment (openbots-ff) [reply OpenBots.Studio/UI/Forms/Supplement Forms/frmPublishProject.cs:111]: done

**Review by openbots-ch — DISMISSED** [review]

**Review by openbots-ab — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]
